os/mm/mm_heap/mm_heapinfo_utils.c

```diff
@@ -0,0 +1,201 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <tinyara/sched.h>
+#include <tinyara/mm/mm.h>
+
+
+/****************************************************************************
+ * Name: heapinfo_add_size
+ *
+ * Description:
+ * Add the allocated size in tcb
+ ****************************************************************************/
+void heapinfo_add_size(struct mm_heap_s *heap, pid_t pid, mmsize_t size)
+{
+	pid_t hash_pid;
+
+	hash_pid = PIDHASH(pid);
+	if (heap->alloc_list[hash_pid].pid == HEAPINFO_INIT_INFO || heap->alloc_list[hash_pid].pid == pid) {
+			heap->alloc_list[hash_pid].pid = pid;
+			heap->alloc_list[hash_pid].curr_alloc_size += size;
+			if (heap->alloc_list[hash_pid].curr_alloc_size > heap->alloc_list[hash_pid].peak_alloc_size) {
+				heap->alloc_list[hash_pid].peak_alloc_size = heap->alloc_list[hash_pid].curr_alloc_size;
+			}
+			heap->alloc_list[hash_pid].num_alloc_free++;
+	}
+}
+
+/****************************************************************************
+ * Name: heapinfo_subtract_size
+ *
+ * Description:
+ * Subtract the allocated size in tcb
+ ****************************************************************************/
+void heapinfo_subtract_size(struct mm_heap_s *heap, pid_t pid, mmsize_t size)
+{
+	pid_t hash_pid;
+
+	hash_pid = PIDHASH(pid);
+	if (heap->alloc_list[hash_pid].pid == pid) {
+			heap->alloc_list[hash_pid].curr_alloc_size -= size;
+			heap->alloc_list[hash_pid].num_alloc_free--;
+	}
+}
+
+/****************************************************************************
+ * Name: heapinfo_update_total_size
+ *
+ * Description:
+ * Calculate the total allocated size and update the peak allocated size for heap
+ ****************************************************************************/
+void heapinfo_update_total_size(struct mm_heap_s *heap, mmsize_t size, pid_t pid)
+{
+	heap->total_alloc_size += size;
+	if (heap->total_alloc_size > heap->peak_alloc_size) {
+		heap->peak_alloc_size = heap->total_alloc_size;
```

Comment:
I can give you same comment as I added top area of this PR

Reply:
Modified. Thanks.